seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogTableUtil.java

```diff
@@ -111,6 +111,16 @@ public static List<CatalogTable> getCatalogTables(
             return Collections.singletonList(catalogTable);
         }
 
+        List<Map<String, Object>> schemaMaps =
+                readonlyConfig.get(TableSchemaOptions.TABLES_CONFIGS);
+        if (schemaMaps != null) {
+            if (schemaMaps.isEmpty()) {
+                throw new SeaTunnelException("tables_configs can not be empty");
+            }
+            return schemaMaps.stream()
+                    .map(map -> buildWithConfig(factoryId, ReadonlyConfig.fromMap(map)))
+                    .collect(Collectors.toList());
+        }
```

Comment:
This is a method marked as @Deprecated. Can it be moved elsewhere?

Reply:
get , the parsing logic is moved to `MongodbIncrementalSourceFactory`